packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx

```diff
@@ -350,20 +352,40 @@ export class DetailsHeaderBase extends BaseComponent<IDetailsHeaderProps, IDetai
       this._dropHintDetails = {};
       this._draggedColumnIndex = -1;
       if (isValidDrop) {
-        this.props.columnReorderOptions!.handleColumnReorder(draggedColumnIndex, targetIndex);
+        if (this.props.columnReorderOptions!.onColumnDrop) {
+          const dragDropDetails: IColumnDragDropDetails = {
+            draggedIndex: draggedColumnIndex,
+            targetIndex: targetIndex
+          };
+          this.props.columnReorderOptions!.onColumnDrop!(dragDropDetails);
+        } else if (this.props.columnReorderOptions!.handleColumnReorder) {
+          this.props.columnReorderOptions!.handleColumnReorder!(draggedColumnIndex, targetIndex);
+        }
       }
     }
   }
 
-  private _setDraggedItemIndex(itemIndex: number) {
+  private _updateDragInfo(itemIndex: number, event?: MouseEvent) {
     if (itemIndex >= 0) {
       // Column index is set based on the checkbox
       this._draggedColumnIndex = this.props.selectionMode !== SelectionMode.none ? itemIndex - 2 : itemIndex - 1;
       this._getDropHintPositions();
-    } else {
+      this.props.columnReorderOptions!.onColumnDragStart!(true);
+    } else if (event && this._draggedColumnIndex >= 0) {
       this._resetDropHints();
       this._draggedColumnIndex = -1;
       this._dropHintDetails = {};
+      const clientRect = this._rootElement!.getBoundingClientRect();
+      if (

```

Comment:
Factor the `if` condition into an `isEventInRegion(event, clientRect)` function and do:
```ts
const {
    onColumnDragEnd
} = this.props;

if (onColumnDragEnd) {
    onColumnDragEnd(event, isEvenInRegion(event, clientRect));
}
```